### PR TITLE
Add coalesce function

### LINKED ISSUE #509: Functions: COALESCE
Add the COALESCE function.

Follow the [PostgreSQL](https://www.postgresql.org/docs/current/functions-conditional.html#FUNCTIONS-COALESCE-NVL-IFNULL) definition.

### PR DESCRIPTION
**Pull request by yaziine**
This PR fixes #509 by adding support to `COALESCE` method.
Unlike PosgreSQL, our implementation doesn't fail if the arguments aren't of the same type.
However, an error is returned if there is no argument at all.

### CODE REVIEW

File: internal/sql/parser/parser.go [hunk 1]
```diff
@@ -268,6 +268,9 @@ func (p *Parser) parseTokens(tokens ...scanner.Token) error {
 // present, it unscans and return false. If the first is present, all the others
 // must be parsed otherwise an error is returned.
 func (p *Parser) parseOptional(tokens ...scanner.Token) (bool, error) {
+	if len(tokens) == 0 {
```
Comment (yaziine) [inline internal/sql/parser/parser.go:271]: This fixes a potential panic if tokens are empty.

File: internal/expr/functions/definition.go [hunk 2]
```diff
@@ -7,6 +7,9 @@ import (
 	"github.com/genjidb/genji/internal/expr"
 )
 
+// UNLIMITED represents an unlimited number of arguments.
+const UNLIMITED = -1
```
Comment (asdine) [inline internal/expr/functions/definition.go:11]: Full uppercase is not really idiomatic. Consider changing this to something like`variadicArity`

File: internal/sql/parser/parser.go (hunk 1 above)
Comment (asdine) [reply internal/sql/parser/parser.go:271]: Panics are good in these cases, we don't want to use this method with empty arguments

**Review by asdine — APPROVED** [review]
👏🏼

**Pull request merged**